pkg/kubelet/kubelet.go

```diff
@@ -2444,7 +2454,9 @@ func (kl *Kubelet) tryUpdateNodeStatus() error {
 		return fmt.Errorf("no node instance returned for %q", kl.nodeName)
 	}
 	kl.networkConfigMutex.Lock()
-	kl.podCIDR = node.Spec.PodCIDR
+	if kl.reconcileCIDR {
+		kl.podCIDR = node.Spec.PodCIDR
```

Comment:
If we expect Spec.PodCIDR to be unset, we could just not clobber kl.podCIDR in that case, and remove the flag.